hack/testing/rsyslog/roles/rsyslog/tasks/main.yaml

```diff
@@ -0,0 +1,159 @@
+---
+- name: Install/Update required packages
+  yum:
+    name: '{{ item }}'
+    state: 'latest'
+  with_flattened:
+    - '{{ rsyslog__base_packages }}'
+    - '{{ rsyslog__viaq_prereq_packages if rsyslog__viaq|bool else [] }}'
+    - '{{ rsyslog__viaq_packages if rsyslog__viaq|bool else [] }}'
```

Comment:
Does this package place the configuration files for viaq?
If yes, where is the repo for it ?

Reply:
The configuration templates are these 2 files.  They are shared among all deployment.
```
https://github.com/nhosoi/origin-aggregated-logging/blob/rsyslog_deploy/hack/testing/rsyslog/roles/rsyslog/templates/etc/rsyslog.conf.j2
https://github.com/nhosoi/origin-aggregated-logging/blob/rsyslog_deploy/hack/testing/rsyslog/roles/rsyslog/templates/etc/rsyslog.d/rules.conf.j2
```
If the question is about the rpm packages, the list is in the defaults/main.yaml:
```
https://github.com/nhosoi/origin-aggregated-logging/blob/rsyslog_deploy/hack/testing/rsyslog/roles/rsyslog/defaults/main.yaml#L56-L69
```
If "the repo" is about the yum repo, it is not maintained in this roles.  (Maybe, I might have misunderstood your questions...?)

Reply:
@sradco https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/rsyslog/mmk8s.conf.j2#L12

This will parse a single line JSON string into individual fields in the record

https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/rsyslog/viaq_formatting.conf#L15

This will parse a JSON valued string field in the record called `$!MESSAGE` into individual fields in the record

Here is the parse_json rulebase: https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/rsyslog/parse_json.rulebase

In general, rsyslog tries to avoid using regexp to parse, because regexp are slow and expensive.  Instead, they have their own high performance log parsing and normalizing library liblognorm http://www.liblognorm.com/ and their own high performance json parsing and formatting library https://github.com/rsyslog/libfastjson/

The rest of the file https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/rsyslog/viaq_formatting.conf shows how to convert journald and kubernetes file logs into the viaq common data model format.

For elasticsearch - https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/rsyslog/elasticsearch.conf.j2#L55
The template field controls how the record is formatted into the JSON blob to send in the bulk index request
```
 template="viaq_template"
```
The template is defined here: https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/rsyslog/elasticsearch.conf.j2#L1
The `$!all-json-plain` is a special property which means "format all of the message fields in $! as a json string in the usual JSON format".